### PR TITLE
Fix projection matrix for raster image drawables

### LINKED ISSUE #2189: [iOS] MLNRasterStyleLayer gets cut off with mask in StyleURL file since v6.0
**Describe the bug**
MLNRasterStyleLayer is cut off/masked by the mask defined in StyleURL JSON file, even though it should not.

See demo app: [MapLibreMaskBug on Github](https://github.com/NicolasBrunnerSRF/mapLibreMaskBug)

This demo app sets a custom style to the MapView, which contains a mask to cut out Switzerland and mask all other countries. It then adds an ImageLayer on top of the map (to show precipitation data). This ImageLayer should not be cut/masked outside of Switzerland. This worked fine with the latest major release of MapLibre v5 but is broken since the introduction of v6. See ReadMe of demo app github repo for a detailed description.

**To Reproduce**
1. Download demo app
2. Build and run on simulator

**Expected behavior**
ImageLayer (MLNRasterStyleLayer) is not affected by StyleURL file.

**Screenshots**
See demo app github repo ReadMe.

**Platform information (please complete the following information):**
 - iOS 17.4
 - XCode 15.3
 - MapLibre iOS v6.2.0 (was fine in v5)

**Additional context**
We are currently in a deadlock situation in our project: Using MapLibre iOS v5 crashes our app on iOS 17.4 simulators with XCode 15.3. Also, we cannot use MapLibre iOS v6 due to the above bug. So we are currently stuck at using XCode 15.2

### PR DESCRIPTION
**Pull request by stefankarschti**
Fixes the projection matrix for raster image drawables that are not tile based. This results in correct depth output according to the layer index and sub-index.

Fixes #2189

### CODE REVIEW

**Review by mwilsnd — APPROVED** [review]

**Comment by sjg-wdw** [comment]
Nice!  How hard would it be to get some test coverage on this?

**Review by sjg-wdw — APPROVED** [review]

**Comment by stefankarschti** [comment]
> Nice! How hard would it be to get some test coverage on this?

I'm looking into that.

**Review by louwers — APPROVED** [review]
Nice one!

File: metrics/cache-style.db
Comment (TimSylvester) [inline metrics/cache-style.db:1]: Is this supposed to be in git?

**Review by TimSylvester — APPROVED** [review]

**Review by alexcristici — APPROVED** [review]

File: metrics/cache-style.db
Comment (louwers) [reply metrics/cache-style.db:1]: It already is.

Comment (stefankarschti) [reply metrics/cache-style.db:1]: yes, I needed to add the radar2.png image referenced from the [style.json](https://github.com/stefankarschti/maplibre-native/blob/issue-2189/metrics/integration/render-tests/combinations/fill-opaque--image-translucent/style.json)

Comment (TimSylvester) [reply metrics/cache-style.db:1]: Not the images, the `cache-style.db`.  I can't tell at a glance what it's for, but generally cache files shouldn't be kept around or transported between clients.  Seems like that might cause misleading results.  I see that you didn't add it, but I wonder why it changed and whether that change is important to capture, or if perhaps it should just be deleted.

Comment (stefankarschti) [reply metrics/cache-style.db:1]: I agree that would be good to reexamine. Initially I copied `radar2.png` to `images/` but the test runner didn't manage to load it, therefore I found that it got the resources from that sqlite database and inserted there. 
As a side note, the fields contain additional information like expiration date, etc.

Comment (louwers) [reply metrics/cache-style.db:1]: It is intentional that the cache is checked in. The render tests run offline.

Comment (TimSylvester) [reply metrics/cache-style.db:1]: ah, ok

**Pull request merged**